From aaaf3251925bc4eac5f3fe7c5d02da220038761c Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Sat, 1 Jul 2023 10:25:10 +1100
Subject: [PATCH 07/10] unpin gast dependency for pip_package

Commit c762c4501ca017994c1fa5554c3c8e47b7c80b66 noted that
"I also know that gast 0.5.2 breaks one of our tests" and committed
"gast versions above 0.4.0 are incompatible with some of TF's tests."

In #56244, it was suggested to raise a PR to remove the cap. In view
of the above-mentioned constraints, I've excluded everything after 0.4.0
up to 0.5.2; note that 0.5.0 directly followed 0.4.0, see
https://github.com/serge-sans-paille/gast/tags
---
 tensorflow/tools/pip_package/setup.py | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/tensorflow/tools/pip_package/setup.py b/tensorflow/tools/pip_package/setup.py
index f7ab72c4..87acb041 100644
--- a/tensorflow/tools/pip_package/setup.py
+++ b/tensorflow/tools/pip_package/setup.py
@@ -85,8 +85,7 @@ REQUIRED_PACKAGES = [
     'absl-py >= 1.0.0',
     'astunparse >= 1.6.0',
     'flatbuffers >= 23.1.21',
-    # TODO(b/213222745) gast versions above 0.4.0 break TF's tests
-    'gast >= 0.2.1, <= 0.4.0',
+    'gast >=0.2.1,!=0.5.0,!=0.5.1,!=0.5.2',
     'google_pasta >= 0.1.1',
     'h5py >= 2.9.0',
     'numpy >= 1.22',
-- 
2.39.2 (Apple Git-143)

